lib/SimpleSAML/Utils/Config.php

```diff
@@ -55,4 +55,21 @@ public static function getSecretSalt()
 
         return $secretSalt;
     }
-}
\ No newline at end of file
+
+    /**
+     * Returns the path to the config dir
+     * If the SIMPLSAMLPHP_CONFIG_DIR environment variable has been set, it takes precedence
+     * over the default $simplesamldir/config directory.
+     *
+     * @return string
+     */
+    public static function getConfigDir()
+    {
+        $configDir = getenv('SIMPLESAMLPHP_CONFIG_DIR');
+        if ($configDir === false || !is_dir($configDir)) {
```

Comment:
If `SIMPLESAMLPHP_CONFIG_DIR` is set to some path which does not exist, it will fall back to another config dir silently. This seems like a source of big confusion. Maybe better that if someone sets that env and it's invalid, to throw an error?